src/errorpage.h

```diff
@@ -93,6 +93,7 @@ class ErrorState
 public:
     /// creates an error of type other than ERR_RELAY_REMOTE
     ErrorState(err_type type, Http::StatusCode, HttpRequest * request, const AccessLogEntryPointer &al);
+    ErrorState(err_type type, Http::StatusCode status, char const *url, Ip::Address &src_addr, HttpRequest *request, const AccessLogEntry::Pointer &al);
```

Comment:
It is not clear when I should use the old documented constructor and when the new undocumented one. Please add a source code comment to describe the new constructor with this "constructor choice" problem in mind.

Reply:
I'm leaving a TODO to consolidate constructors onto a single one to avoid scope creep on this PR

Reply:
Please be sure to use the word 'deprecated' in the docs for the one that should be removed (IMO that is the one with url parameter).

Reply:
Yes

Reply:
> Please ping me when the PR is in a reviewable state again and clientBuildError-related changes are really undone (except for moving the function to another file).

FYI: The first precondition has not been met yet -- GitHub diff still shows lots of unrelated changes/noise -- so I could not check clientBuildError-related changes and bounced the PR back to your side.